crates/modules/src/applications/ics29_fee/events.rs

```diff
@@ -0,0 +1,95 @@
+use core::str::FromStr;
+use itertools::Itertools;
+use serde_derive::{Deserialize, Serialize};
+use tendermint::abci::tag::Tag;
+use tendermint::abci::Event as AbciEvent;
+
+use super::error::Error;
+use crate::applications::transfer::coin::RawCoin;
+use crate::core::ics04_channel::packet::Sequence;
+use crate::core::ics24_host::identifier::{ChannelId, PortId};
+use crate::events::IbcEventType;
+use crate::prelude::*;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct IncentivizedPacket {
+    pub port_id: PortId,
+    pub channel_id: ChannelId,
+    pub sequence: Sequence,
+    pub total_recv_fee: Vec<RawCoin>,
+    pub total_ack_fee: Vec<RawCoin>,
+    pub total_timeout_fee: Vec<RawCoin>,
```

Comment:
What does `total` denote in this case? Is it that an `IncentivizedPacket` can contain multiple fees?

Reply:
It means the total fees that has been paid to the packet. Yes the fees can be paid multiple times.